src/libvncserver/websockets.c

```diff
@@ -289,13 +291,15 @@ webSocketsHandshake(rfbClientPtr cl, char *scheme)
         return FALSE;
     }
 
-    if ((protocol) && (strstr(protocol, "base64"))) {
+    proto_binary = strstr(protocol, "binary");
+    proto_base64 = strstr(protocol, "base64");
+    if ((protocol) && (proto_base64 && (!proto_binary || proto_base64 < proto_binary))) {
```

Comment:
After the `||`, you're comparing `char*`? That can't be right.

Reply:
Comparing pointers is the priority check: the one that comes first has the priority.